File: AtCoder_Beginners_Selection/ABC085B/main.py
def main():
    N: int = int(input()) # N枚の円形の餅
    d_list: set[int] = set()

    for _ in range(0, N):
        d_list.add(int(input())) # [15 12 13]

    # input check
    assert(1 <= N <= 100)
    for i in d_list:
        assert(1 <= i <= 100)

    print(len(d_list))
if __name__ == "__main__":
    main()

File: AtCoder_Beginners_Selection/ABC085B/main.nim
import strutils
import sets
proc main =
    let N: int = parseInt(stdin.readLine())
    var d_list: seq[int]
    for _ in 0 ..< N:
        d_list.add(parseInt(stdin.readLine()))

    # input check
    assert(1 <= N and N <= 100)
    for i in d_list:
        assert(1 <= i and i <= 100)

    let result = toHashSet(d_list)
    echo(len(result))

when isMainModule:
    main()
